test-network-function/lifecycle/suite.go

```diff
@@ -104,9 +108,65 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 		}
 
 		testOwner(&configData)
+
+		testScaling(&configData)
 	}
 })
 
+func waitForAllDeploymentsReady(namespace string, timeout, pollingPeriod time.Duration) {
+	gomega.Eventually(func() []string {
+		_, notReadyDeployments := getDeployments(namespace)
```

Comment:
Eventually we should probably replace this with a single deployment query. It might be safe to assume that we own the whole namespace in our lab, but not so much for on-prem test ran by customers.

Reply:
Ok.